src/configs.ts

```diff
@@ -15,14 +15,48 @@ const configFilenames = [
   'eslint.config.cts',
 ]
 
+export interface ReadConfigOptions {
+  cwd: string
+  userConfigPath?: string
+  userRootPath?: string
+  /**
+   * Change current working directory to rootPath
+   * @default true
+   */
+  chdir?: boolean
+}
+
 export async function readConfig(
-  cwd: string,
-  configPathOverride = process.env.ESLINT_CONFIG,
+  {
+    cwd,
+    userConfigPath,
+    userRootPath,
+    chdir = true,
+  }: ReadConfigOptions,
 ): Promise<{ payload: Payload, dependencies: string[] }> {
-  const configPath = resolve(cwd, configPathOverride || configFilenames.find(i => fs.existsSync(resolve(cwd, i))) || configFilenames[0])
+  if (userRootPath)
+    userRootPath = resolve(cwd, userRootPath)
+
+  const configPath = userConfigPath
+    ? resolve(cwd, userConfigPath)
+    : await findUp(configFilenames, { cwd: userRootPath || cwd })
```

Comment:
Since the flat config is just a JS module, I found it's great that you can just import it directly to get the resolved files. While I pretty much want to share the code, I don't want it to be coupled with users' ESLint version, or specific API that ESLint made public. Right now, the config searching logic isn't complicated; I guess it's ok not to keep it standalone and portable right now. For sure, I am still open to sharing logic if needed in the future.

Reply:
Fair enough. My only concern is keeping it in sync if things change, but if we launch from ESLint we can always pass in the specific config file and base path, so maybe not that big of an issue.